src/Components/VoteArticle.jsx

```diff
@@ -0,0 +1,38 @@
+import { useState } from "react";
+import { voteArticle } from "./Api";
+export default function VoteArticle({ articleId, initialVotes }) {
+  const [voteCount, setVoteCount] = useState(initialVotes);
+  const [voted, setVoted] = useState(false);
+  const [error, setError] = useState(null);
+
+  const handleVote = (increment) => {
+    if (!voted) {
+      voteArticle(articleId, increment)
+        .then(() => {
+          setVoteCount((prevVoteCount) =>
+            increment===1 ? prevVoteCount + 1 : prevVoteCount - 1
+          );
+          setVoted(true);
+        })
+        .catch((error) => {
+          setError(error);
+        });
+    }
+  };
+
+  if (error) {
+    return <div>Error: {error.message}</div>;
```

Comment:
It's really good to have this consideration of error handling, something else to consider is that the error you get back is intended for developers. So it might not always be the most user friendly to an average website consumer to read